docs/usage/directives/directives-file-syntax.md

```diff
@@ -0,0 +1,159 @@
+---
+sidebar_position: 2
+---
+# Directives File Syntax
+
+The directives file syntax heavily relies on the *INI config* syntax, but with several modifications tailored to suit huronOS' requirements. As explained in the [introduction to directives](introduction-to-directives), the directives are primarily governed by their *execution modes* (*contest*, *event*, *default*). However, some directives, such as keyboard distributions and timezone settings, are considered mode-independent, while others are mode-dependent. The directives are divided into two sections: Global Directives and Specific Directives.
+
+For specific directives, we have three sections (case-sensitive):
+- "**Always**" for the default mode.
+- "**Event**" for the event mode.
+- "**Contest**" for the contest mode.
+
+When dealing with *Event* and *Contest* modes, we also expect the sections "**Event-Times**" and "**Contest-Times**" (respectively) to be present, resulting in a file structure similar to this:
+
+```ini
+[Global]
+
+[Always]
+
+[Event]
+
+[Contest]
+
+[Event-Times]
+
+[Contest-Times]
+
+```
+
+## Global Directives
+
+The global directives, applied at all times, support various configurations. Each specific syntax is detailed in its own [config documents](./configurations/). This section includes the following directives (click on the title link to view the specific syntax documentation):
+
+- [`TimeZone=<YourTimezone>`](./configurations/timezones.md)  
+    This directive allows you to set the system's local timezone using IANA standard timezones.
+
+- [`ConfigExpirationTime=[ never || ISO8601 Time ]`](./configurations/expiration-time.md)  
+    Use this directive to ensure that a set of rules (directives file) expires after a certain time. You can set an ISO 8601 time-date to specify the expiration time.
+
+- [`AvailableKeyboardLayouts=[layout1|...|layoutN|]`](./configurations/keyboard-layout.md)  
+    Here, you can define XKB keyboard layouts available for quick switching via the system tray button. Other configurations can be set via the system configuration. A list containing the possible layouts is available [here](https://gist.github.com/jatcwang/ae3b7019f219b8cdc6798329108c9aee).
+
+- [`DefaultKeyboardLayout=layout`](./configurations/keyboard-layout.md)  
+    This directive allows you to specify the XKB keyboard layout to be set as the default on each mode change or at boot.
```

Comment:
And what if no default layout is specified / an invalid one is defined?
Is there a fallback layout in place?

Reply:
> And what if no default layout is specified / an invalid one is defined? Is there a fallback layout in place?

Hahaha, this one is funny. So if I set the keyboard layout 'cacuca' it actually sets it! But gnome itself will do the fallback to US keyboard if the name is not valid. This is because in theory anyone can create it's custom XKB, so any id could be possible in theory. 
In practice, this was supposed to be cached by the old directives validator (https://github.com/equetzal/huronOS-build-tools/issues/157), which is not working anymore. Another reason to prioritize it.